### DISCUSSION TITLE
Impossible to continue using kit

### DISCUSSION THREAD

**Discussion by NormandoHall** [sveltejs/kit#6679] [General]
After 405, I really unable to follow the breaking changes. There are a lot of missing examples to reference the new changes, and the "realworld" is outdated, and I can't figure how to do a simple migration of session, cookie based.

"load" everywhere, "$page.locals", "cookie", "event, now "no event". It is a madness. I stop losing my time right now.

**Comment by richarddavenport** [comment]
I was in the same boat... was on 405. I scaffolded out a new SvelteKit app and started test driving the new way things work. OMG! I love it. Once you get past the migration, I think you'll love how well typescript is integrated, the DX is overall a better experience. Forms is really nice now as well.

**Reply by NormandoHall** [reply]
@richarddavenport  I hope I feel the same, but at the moment I'm finding it frustrating, especially dealing with session and cookies. Many up-to-date examples are missing to pick up and learn apart from the bare documentation. In the meantime I'll leave Kit and come back to it when it's something better documented, and stable. Cheers

**Comment by teemingc** [comment]
Have you read the [migration guide](https://github.com/sveltejs/kit/discussions/5774)?

**Reply by NormandoHall** [reply]
> Have you read the [migration guide](https://github.com/sveltejs/kit/discussions/5774)?

Yes, of course, and run it for routes and it is ok. But there are also multiple changes, session is removed, etc etc. The session removed is a big issue for me. I will wait for real examples to learn from they.

**Reply by teemingc** [reply]
> The session removed is a big issue for me. I will wait for real examples to learn from they.

Personally, I've found session to be buggy before it was removed, so I stopped using it long ago.

However, you may be interested in this guide that details how to create your own `session`
https://www.captaincodeman.com/re-creating-the-sveltekit-session-store

**Reply by NormandoHall** [reply]
Thanks @s3812497 I will try when kit becomes more stable.

**Reply by binajmen** [reply]
Is there a proper example on _vanilla_ cookie management available out there? I've the same conclusion than @NormandoHall at the moment: pretty hard to embrace the changes without proper examples. Pretty sure we'll see a ton of tuts when 1.0 is finally released, but there are needs now too :/

**Reply by teemingc** [reply]
> Is there a proper example on _vanilla_ cookie management available out there? I've the same conclusion than @NormandoHall at the moment: pretty hard to embrace the changes without proper examples. Pretty sure we'll see a ton of tuts when 1.0 is finally released, but there are needs now too :/

Not sure what you mean by vanilla, but in SvelteKit:

A `+page.server.js` and `+layout.server.js` file receives a `cookies` input and can be used such as:

```js
// +page.server.js
export const load = ({ cookies }) => {
  // gets a value from a cookie
  const cookie_value = cookies.get('your_cookie_name')
  
  // adds a set-cookie header with an expired cookie to delete it
  cookies.delete('your_cookie_name')
  
  // adds to the set-cookie header to add a new cookie
  cookies.set('new_cookie_name', 'some-value')
}
```

https://kit.svelte.dev/docs/types#sveltejs-kit-cookies

**Reply by binajmen** [reply]
Actually, I was thinking about the authentication process. (It seems to me) There are tons of concept to integrate to implement it properly, yet no available (at my knowledge) documentation on how to do it. Packages that are providing such services (Lucia, etc.) are suffering from these breaking changes, so I'm not even sure I can trust them at the moment.

As a newcomer, I'm less impacted (and disturbed) by these breaking changes. I started to play with SK 1 week ago for a side (yet need-to-be-production-ready) project within my company, so I'm clearly enjoying whatever I got at that time, but reading all the disappointments and having hard time grasping basic concepts leads me to believe SK is not yet ready. I might have to go back to Remix for this one and just come back to SK when 1.0 is out.

**Reply by NormandoHall** [reply]
@s3812497  I appreciate your effort and help. I know the cookie type that is available, but we need a complete example, like "realworld" that implements all the handling of the session in hooks, etc, the login with the new form, etc etc. Nothing is safe, we don't even know the best way to implement a full cookie-based authentication.

I agree with @binajmen comments. I have been developing with SK for more than 9 months, and at this point I feel frustrated, and even unsure of my own knowledge. That makes me feel SK

**Reply by teemingc** [reply]
> Actually, I was thinking about the authentication process.

For myself, I have a function that checks for an access token inside a cookie then validates it, returning true/false.
I use this in the [`load` function](https://kit.svelte.dev/docs/load) of `+page.server.ts` or `+layout.server.ts` (for multiple pages) and redirect to the login page if the access token is missing / invalid.

For logins, I use an [`action`](https://kit.svelte.dev/docs/form-actions) in the `+page.server.ts` of the login route that returns a cookie with the access token on successful login.

> reading all the disappointments and having hard time grasping basic concepts leads me to believe SK is not yet ready.

Personally, I've found the new changes very helpful, and don't really agree with the disappointment.

**Reply by binajmen** [reply]
Although I could find a way to make it work (most probably by following your process, trying and failing), when picking a framework such as Remix, Next or SvelteKit, the goal is to have such things already prepared to a maximum, well documented, so I can implement it when needed, allowing me to focus on the real business value of the application.

It's been one week I've been exploring SK in my spare time to see if this could be the right candidate, but so far I don't feel confident  I'll be able to reach any deadlines.. I believe in my case the current state of SK is a deal breaker.

(But as you, and as a newcomer that don't have to migrate every months his app to cope with these breaking changes, I like the direction it takes, and will certainly come back and try SK when 1.0 is published)

**Reply by binajmen** [reply]
RC1 submitted =D

**Comment by palbergstrom** [comment]
I have the same feeling. I try to follow the idea and understand the changes but the new way makes it more complicated, not least visually. Perhaps a new theme plugin in VS Code that can help with that, make every directory that is a route stand out. 

Why fix what isn't broken or try to reinvent the wheel? The dev team is way beyond me in knowledge but I have a hunch they will have to take a step back or slowly loose momentum in popularity. But what do I know. 

Edit: Getting more used to it and Form Actions helps.

**Reply by teemingc** [reply]
Which changes are you having trouble adapting to?

**Reply by palbergstrom** [reply]
Routing. Would be better if they kept the file based routing but instead have directories +server and +load. /about.svelte would have the same name in server and load.

**Reply by apexfactory** [reply]
> Routing. Would be better if they kept the file based routing but instead have directories +server and +load. /about.svelte would have the same name in server and load.

Ew.

**Reply by palbergstrom** [reply]
@apex-apps-ltd What does Ew stand for?
